crates/sui/src/client_ptb/ptb.rs

```diff
@@ -0,0 +1,680 @@
+            .map(|(sa, alias)| (alias.alias.clone(), AccountAddress::from(*sa)))
+            .collect();
+        let mut builder = PTBBuilder::new(starting_addresses, client.read_api());
+
+        for p in parsed.into_iter() {
+            builder.handle_command(p).await;
+        }
+
+        builder.finish()
+    }
+
+    // This function is used to split the input string into arguments. We decide when we have a new
+    // argument based on the different delimiters that we have. If we are inside of a delimiter and
+    // we hit a space, we should keep going until we find the end of the delimiter and a space at
+    // which point that is the end of the argument.
+    fn split_into_args(s: String) -> Vec<String> {
+        let mut res = vec![];
+        let mut temp = String::new();
+
+        // Argument delimiters that cannot span multiple arguments. We know if we are inside of one
+        // of these that we need to keep going to finish the argument.
+        let mut in_quotes = false;
+        let mut in_ticks = false;
+        let mut brackets = 0;
+        let mut parens = 0;
+        let mut hairpins = 0;
+
+        for c in s.chars() {
+            if c == '"' {
+                in_quotes = !in_quotes;
+            }
+            if c == '\'' {
+                in_ticks = !in_ticks;
+            }
+
+            // NB: At this point all string escapes have been normalized to single quotes.
+            // When we are in a literal string being passed to the PTB, we do not look or count
+            // delimiters.
+            if !in_ticks {
+                if c == '[' {
+                    brackets += 1;
+                }
+                if c == ']' {
+                    brackets -= 1;
+                }
+
+                if c == '(' {
+                    parens += 1;
+                }
+                if c == ')' {
+                    parens -= 1;
+                }
+
+                if c == '<' {
+                    hairpins += 1;
+                }
+                if c == '>' {
+                    hairpins -= 1;
+                }
+            }
+
+            let is_delimiter = c == ' ' // Hit a whitspace
+                && !in_quotes  // Not currently inside a quote
+                && !in_ticks   // Not currently inside a string literal
+                && brackets == 0 // Not currently inside a bracket (array/vector)
+                && parens == 0  // Not curently inside parens (some)
+                && hairpins == 0; // Not currently inside hairpins (generics)
+
+            if is_delimiter {
+                res.push(temp.clone());
+                temp.clear();
+            } else {
+                temp.push(c);
+            }
+        }
+        res.push(temp);
+        res
+    }
+
+    /// Parses and executes the PTB with the sender as the current active address
+    pub async fn execute(
+        self,
+        matches: ArgMatches,
+        context: Option<WalletContext>,
+    ) -> Result<(), Error> {
```

Comment:
The whole PTB is being resolved and executed from this function, so this is a good point for starting to review.